ddl_checker/main.go

```diff
@@ -0,0 +1,214 @@
+		"Auto mode: The program will automatically synchronize the dependent table structure from MYSQL " +
+		"and delete the conflict table\n" +
+		"Prompt mode: The program will ask you before synchronizing the dependent table structure from MYSQL\n" +
+		"Manual mode: This program does not perform anything other than executing the input SQL.\n\n" +
+		"SETMOD usage: SETMOD <MODCODE>; MODCODE = [\"Auto\", \"Prompt\", \"Manual\"] (case insensitive).\n\n"
+)
+
+func main() {
+	initialise()
+	printWelcome()
+	mainLoop()
+	destroy()
+}
+
+func initialise() {
+	flag.Parse()
+	var err error
+	reader = bufio.NewReader(os.Stdin)
+	executableChecker, err = ddl_checker.NewExecutableChecker()
+	if err != nil {
+		fmt.Printf("[DDLChecker] Init failed, can't create ExecutableChecker: %s\n", err.Error())
+		os.Exit(1)
+	}
+	executableChecker.Execute("use test;")
+	db, err = sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s:%d)/%s",
+		*username, *password, *host, *port, *schema))
+	if err != nil {
+		fmt.Printf("[DDLChecker] Init failed, can't open mysql database: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+}
+
+func destroy() {
+	executableChecker.Close()
+	db.Close()
+}
+
+func mainLoop() {
+	var input string
+	var err error
+	for cont := true; cont; cont = handler(input) {
+		fmt.Printf("[%s] > ", modeName())
+		input, err = reader.ReadString(';')
+		if err != nil {
+			fmt.Printf("[DDLChecker] Read stdin error: %s\n", err.Error())
+			os.Exit(1)
+		}
+	}
+}
+func handler(input string) bool {
+	// cmd exit
+	lowerTrimInput := strings.ToLower(strings.TrimFunc(input, func(r rune) bool {
+		return unicode.IsSpace(r) || r == ';'
+	}))
+	if lowerTrimInput == "exit" {
+		return false
+	}
+	// cmd setmod
+	if strings.HasPrefix(lowerTrimInput, "setmod") {
+		switch strings.TrimSpace(lowerTrimInput[6:]) {
+		case "auto":
+			modCode = 0
+		case "prompt":
+			modCode = 1
+		case "manual":
+			modCode = 2
+		default:
+			fmt.Println("SETMOD usage: SETMOD <MODCODE>; MODCODE = [\"Auto\", \"Prompt\", \"Manual\"] (case insensitive).")
+		}
+		return true
+	}
+	if modCode != 2 {
+		// auto and query mod
+		stmt, err := executableChecker.Parse(input)
+		if err != nil {
+			fmt.Printf("[DDLChecker] SQL parse error: %s\n", err.Error())
+			return true
+		}
+		neededTable := ddl_checker.GetTableNeededExist(stmt)
+		nonNeededTable := ddl_checker.GetTableNeededNonExist(stmt)
+		if modCode == 0 || (modCode == 1 && queryAutoSync(neededTable, nonNeededTable)) {
+			syncTablesFromMysql(neededTable)
+			dropTables(nonNeededTable)
+		}
+	}
+	err := executableChecker.Execute(input)
+	if err == nil {
+		fmt.Println("[DDLChecker] SQL execution succeeded")
+	} else {
+		fmt.Println("[DDLChecker] SQL execution failed:", err.Error())
+	}
+	return true
+}
+
+func syncTablesFromMysql(tableNames []string) {
+	for _, tableName := range tableNames {
+		fmt.Println("[DDLChecker] Syncing Table", tableName)
+		isExist := executableChecker.IsTableExist(tableName)
```

Comment:
maybe not user want？

Reply:
should i ask the user every table?

Reply:
maybe local table is not the table user want, just drop it and create a new one

Reply:
this tool will skip syncing the table what already exist.when user drop the local table,the tool  will just drop it